tests/integration/rosetta/rosetta_test.go

```diff
@@ -0,0 +1,277 @@
+// +build integration
+
+// Copyright 2021 Alvalor S.A.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package rosetta_test
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/dgraph-io/badger/v2"
+	"github.com/klauspost/compress/zstd"
+	"github.com/labstack/echo/v4"
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/flow-go/model/flow"
+
+	"github.com/optakt/flow-dps/api/rosetta"
+	"github.com/optakt/flow-dps/models/dps"
+	"github.com/optakt/flow-dps/models/identifier"
+	"github.com/optakt/flow-dps/service/state"
+
+	"github.com/optakt/flow-dps/rosetta/invoker"
+	"github.com/optakt/flow-dps/rosetta/retriever"
+	"github.com/optakt/flow-dps/rosetta/scripts"
+	"github.com/optakt/flow-dps/rosetta/validator"
+)
+
+const testChainID = "flow-testnet"
+
+// initialize the rosetta Data HTTP handler once and reuse for all tests.
+var rosettaSvc *rosetta.Data
+
+func setupDB() (*rosetta.Data, *badger.DB, error) {
+
+	const testDbBackupFile = "rosetta-get-balance-test.db"
+	opts := badger.DefaultOptions("").
+		WithInMemory(true).
+		WithReadOnly(true).
+		WithLogger(nil)
+
+	db, err := badger.Open(opts)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	dbSnapshot, err := zstd.NewReader(hex.NewDecoder(strings.NewReader(compressedDBSnapshot())))
+	if err != nil {
+		return nil, nil, err
+	}
+
+	defer dbSnapshot.Close()
+
+	err = db.Load(dbSnapshot, 10)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	core, err := state.NewCoreFromDB(db)
+	if err != nil {
+		return nil, nil, err
+
+	}
+
+	// setup scaffolding for minimal server we need for the tests
+	params, ok := dps.FlowParams[flow.ChainID(testChainID)]
+	if !ok {
+		return nil, nil, fmt.Errorf("invalid chain")
+	}
+
+	generator := scripts.NewGenerator(params)
+	invoke := invoker.New(zerolog.Nop(), core)
+	validate := validator.New(params, core.Height())
+	retrieve := retriever.New(generator, invoke)
+
+	svc := rosetta.NewData(validate, retrieve)
+
+	return svc, db, err
+}
+
+func TestMain(m *testing.M) {
+
+	svc, db, err := setupDB()
+	if err != nil {
+		log.Fatalf("could not perform setup for test: %v", err)
+	}
+
+	rosettaSvc = svc
```

Comment:
I really dislike the use of the global variable here. As mentioned, please look into testify test suites instead, they manage this kind of test setup and teardown for integration tests more cleanly.